Harvest.Core/Services/SlothService.cs

```diff
@@ -259,6 +173,127 @@ public async Task<Result<SlothResponseModel>> MoveMoney(int invoiceId)
             return Result.Error("Sloth Response didn't have a success code for moneyTransfer id {transferId}: {data}", invoice.Id, data);
         }
 
+        private async Task<bool> ProcessDebits(TransactionViewModel model, decimal grandTotal, Invoice invoice)
+        {
+            //Passthrough rates have their own object codes, so they should be split properly here.
+            var expenseGroups = invoice.Expenses
+                .GroupBy(a => a.ExpenseObjectCode)
+                .Select(s => new { objectCode = s.Key, total = s.Sum(s1 => s1.Total) })
+                .ToArray();
+
+            var validatedProjectAccounts = new Dictionary<Account, AccountValidationModel>();
+            //Validate accounts
+            foreach (var projectAccount in invoice.Project.Accounts)
+            {
+                var account = await _financialService.IsValid(projectAccount.Number);
+                if (!account.IsValid)
+                {
+                    Result.Error("Unable to validate debit account {debitAccount}: {debitMessage}",
+                        projectAccount.Number, account.Message);
+                    return false;
```

Comment:
Not a big deal, but why not have this method return a `Task<Result<bool>>` and return these errors? Then the calling method could just pass along the error instead of creating its own `"ProcessDebits failed"` message.

Reply:
I wasn't sure what that would look like to get it to log the error and return out of the calling method.